crates/oxc_linter/src/rules/eslint/no_redundant_constructor_init.rs

```diff
@@ -0,0 +1,165 @@
+use oxc_ast::{
+    AstKind,
+    ast::{AssignmentTarget, Expression, MethodDefinitionKind, Statement},
+};
+use oxc_diagnostics::OxcDiagnostic;
+use oxc_macros::declare_oxc_lint;
+use oxc_span::Span;
+
+use crate::{AstNode, context::LintContext, rule::Rule};
+
+fn no_redundant_constructor_init_diagnostic(span: Span) -> OxcDiagnostic {
+    OxcDiagnostic::warn("Explicit initialization of public members is redundant")
+        .with_help("Remove the explicit initialization")
+        .with_label(span)
+}
+
+#[derive(Debug, Default, Clone)]
+pub struct NoRedundantConstructorInit;
+
+declare_oxc_lint!(
+    /// ### What it does
+    ///
+    /// Prevents redundant initialization of class members within a constructor.
+    ///
+    /// ### Why is this bad?
+    ///
+    /// Arguments marked as `public` within a constructor are automatically initialized.
+    /// Providing an explicit initialization is redundant and can be removed.
+    ///
+    /// ### Examples
+    ///
+    /// Examples of **incorrect** code for this rule:
+    /// ```js
+    /// class Foo {
+    ///   constructor(public name: unknown) {
+    ///     this.name = name;
+    ///   }
+    /// }
+    /// ```
+    ///
+    /// Examples of **correct** code for this rule:
+    /// ```js
+    /// class Foo {
+    ///   constructor(public name: unknown) {}
+    /// }
+    /// ```
+    NoRedundantConstructorInit,
+    eslint,
+    style,
```

Comment:
This is definitely a `correctness` rule, because the offending code is useless, or a mistake.

Reply:
Thanks for clarifying